### PR TITLE
ddl: Add index fast path to be a default add index path

### LINKED ISSUE #35983: Improve the performance of adding index
## Feature Request

**Is your feature request related to a problem? Please describe:**

Adding indexes is a high-frequency operation in the proof-of-concept scenarios. However, for the large tables, it is a time-consuming process that may affect the user experience. There is still room for improvement in TiDB. For the similar requests, see #12963 and #14454.

**Describe the feature you'd like:**

We would like to achieve a better performance for adding index in TiDB, and reduce the impact on DML transactions for foreground traffic. 

## Stage 1: support the ingest method (3X)

~~Feature branch: #36767~~

Design:
- [x] #35196

Refactoring:
- [x] #36085
- [x] #36091
- [x] #37507
- [x] #37716

Config:
- [x] #36088
- [x] #37849

Framework:
- [x] #37271
- [x] #37521
- [x] #37709
- [x] #37847

Tests:
- [x] #37320
- [x] #37782
- [x] #37879
- [x] #37911

## Stage 2: optimize add index reorganization (10X)

~Experimental branch: https://github.com/pingcap/tidb/pull/38110~

- [x] #38646
- [x] #39191
- [x] #39249
- [x] #39267
- [x] #39345

### PR DESCRIPTION
**Pull request by Benjamin2037**
### What problem does this PR solve?

Issue Number: ref #35983

Problem Summary:
This PR will make  tidb_ddl_enable_fast_reorg default and make Add index fast path to be default solution for add index.

### Check List

Tests 

- [X] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by jackysp — APPROVED** [review]
LGTM

File: ddl/cancel_test.go [hunk 1]
```diff
@@ -232,6 +232,8 @@ func TestCancel(t *testing.T) {
 
 	// Prepare schema.
 	tk.MustExec("use test")
+	// ToDo: will check why tidb_ddl_enable_fast_reorg could not be default on in another pr.
```
Comment (zimulala) [inline ddl/cancel_test.go:235]:
```suggestion
	// TODO: Will check why tidb_ddl_enable_fast_reorg could not be defaulted on in another PR.
```

Please update other same comments.

File: ddl/db_change_test.go [hunk 2]
```diff
@@ -1719,6 +1719,8 @@ func TestCreateUniqueExpressionIndex(t *testing.T) {
 
 	tk := testkit.NewTestKit(t, store)
 	tk.MustExec("use test")
+	// ToDo: will check why tidb_ddl_enable_fast_reorg could not be default on in another pr.
```
Comment (Benjamin2037) [inline ddl/db_change_test.go:1722]:
```suggestion
// TODO: Will check why tidb_ddl_enable_fast_reorg could not be defaulted on in another PR.
```

File: ddl/failtest/fail_db_test.go [hunk 3]
```diff
@@ -320,6 +320,8 @@ func TestRunDDLJobPanicEnableClusteredIndex(t *testing.T) {
 	s := createFailDBSuite(t)
 	testAddIndexWorkerNum(t, s, func(tk *testkit.TestKit) {
 		tk.Session().GetSessionVars().EnableClusteredIndex = variable.ClusteredIndexDefModeOn
+		// ToDo: will check why tidb_ddl_enable_fast_reorg could not be default on in another pr.
```
Comment (Benjamin2037) [inline ddl/failtest/fail_db_test.go:323]:
```suggestion
// TODO: Will check why tidb_ddl_enable_fast_reorg could not be defaulted on in another PR.
```

**Review by zimulala — APPROVED** [review]

**Pull request merged**